pkg/network/telemetry/debugfs_stat_collector_linux.go

```diff
@@ -0,0 +1,92 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build !windows
+
+package telemetry
+
+import (
+	manager "github.com/DataDog/ebpf-manager"
+	"github.com/DataDog/ebpf-manager/tracefs"
+	"regexp"
+
+	"github.com/DataDog/datadog-agent/pkg/ebpf"
+	"github.com/prometheus/client_golang/prometheus"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const kProbeTelemetryName = "ebpf__kprobes"
+
+// event name format is p|r_<funcname>_<uid>_<pid>
+var eventRegexp = regexp.MustCompile(`^((?:p|r)_.+?)_([^_]*)_([^_]*)$`)
+
+var myPid int
+
+func init() {
+	myPid = manager.Getpid()
+}
+
+type DebugFsStatCollector struct {
+	hits   *prometheus.Desc
+	misses *prometheus.Desc
+}
+
+func NewDebugFsStatCollector() *DebugFsStatCollector {
+	return &DebugFsStatCollector{
+		hits:   prometheus.NewDesc(kProbeTelemetryName+".hits", "Gauge tracking number of kprobe hits", nil, nil),
```

Comment:
Shouldn't these be counters?

Reply:
these track the live value of the metric, so they can either be gauges or if we want them to be counters, we can do what we've done in other cases and track the delta between this value and the last measurement

Reply:
If the live value is monotonically increasing, I think it makes more sense to use counters.

Reply:
https://github.com/DataDog/datadog-agent/pull/15249#discussion_r1165856026 it seems bryce had a different take on these originally, and I think we've gone back and forth on this a few times

@hmahmood @brycekahle if a value is monotonically increasing should it be:
- a gauge set from the live value
- a counter but we track the last value and increment by the delta of the prev and current (in this case a bit more complex because there are many probes)

Reply:
My comment was that you couldn't directly use the value for a counter. If you compute a delta, then a counter is fine.